### PR TITLE
Add dynamic loading of changes of config (only in onvif divices) file

### LINKED ISSUE #2: Dynamic loading of changes of config.yml
Hi!
Great job! This is what I was looking for a very long time.
Is it possible to add config.yml monitoring?
The task is to dynamically add/remove camera's subscriptions without restarting container
Thanks!

### PR DESCRIPTION
**Pull request by benedikt45**
If add/remove/change device - it will be dynamic reloading
Issued #2  

Close #2

### CODE REVIEW

File: src/Config.js [hunk 1]
```diff
@@ -15,14 +16,16 @@ class Config {
 
   _config;
 
-  constructor() {
```
Comment (dmitrif) [inline src/Config.js:18]: Since we are currently using Config only in one place (the manager) and it is being changed to be invoked with a constructor parameter, we should remove the Singleton plumbing below (ex. `if (Config.instance)` etc.

File: src/Config.js [hunk 2]
```diff
@@ -77,7 +80,23 @@ class Config {
     return mergedConfig;
   };
 
+  onConfigChange = (callback) => {
+    const watcher = chokidar.watch([DEFAULT_CONFIG_PATH, CONFIG_PATH]);
+    watcher.on('change', async () => {
+      let tempConfig = this._loadConfig();
+      // Check if onvif device config changed. Only this change observe
+      if (tempConfig?.onvif !== undefined && this._config?.onvif !== undefined) {
```
Comment (dmitrif) [inline src/Config.js:88]: Seeing as `onvif` is either an object or undefined, maybe we can shorten this to: 
```
      if (tempConfig?.onvif && this._config?.onvif) {
```

File: src/Config.js [hunk 3]
```diff
@@ -77,7 +80,23 @@ class Config {
...
+        let temp = await JSON.stringify(tempConfig.onvif),
+            current =  await JSON.stringify(this._config.onvif);
+        if (temp!== current) {
```
Comment (dmitrif) [inline src/Config.js:91]: I suggest we use `lodash.isEqual` here to check for deep equality as the following is not equal:
image

File: src/Manager.js [hunk 4]
```diff
@@ -14,24 +14,30 @@ const convertBooleanToSensorState = bool => bool ? 'ON' : 'OFF';
...
-    this.initializeOnvifDevices(config.get('onvif'));
-    this.subscriber.withCallback(CALLBACK_TYPES.motion, this.onMotionDetected);
+    this.initializeOnvifDevicesFunctions();
 
     this.onExitSendStatus();
-};
+  };
+
+  initializeOnvifDevicesFunctions = () => {
+    this.subscriber = new OnvifSubscriberGroup();
```
Comment (dmitrif) [inline src/Manager.js:37]: I think we may need to add some teardown / deconstructor logic to remove old event subscribers if an OnvifSubscriberGroup already exists to avoid duplicate events between garbage collection cycles.

**Review by dmitrif — CHANGES_REQUESTED** [review]
Thanks a lot! This is looking great, just had some feedback on a few things.

**Comment by benedikt45** [comment]
Fix all remarks in commits

File: src/Manager.js [hunk 5]
```diff
@@ -67,7 +74,7 @@ export default class Manager {
   };
 
   /* Event Callbacks */
-  onMotionDetected = debounceStateUpdate((onvifDeviceId, boolMotionState) => {
+    onMotionDetected = debounceStateUpdate((onvifDeviceId, boolMotionState) => {
```
Comment (dmitrif) [inline src/Manager.js:77]: Whitespace added by mistake I think!

**Review by dmitrif — APPROVED** [review]
LGTM minus whitespace nitpick.
Thank you!

File: src/Manager.js (hunk 5 above)
Comment (benedikt45) [reply src/Manager.js]: Yep, mistake, fix

**Comment by dmitrif** [comment]
@all-contributors please add @benedikt45 for code

**Comment by dmitrif** [comment]
Merged, thank you! Will release after cleaning up docs.
